src/pages/guides/reference/configuringproxy.md

```diff
@@ -1,7 +1,96 @@
+Runtime does not expose egress IPs due to security reasons. If customers need a way to securely communicate with their backend services, they can use a proxy in between their backend service and I/O Runtime.
 
-This can be done using the following solution: 
+This can be done by the addition of a proxy component (in this example, an AWS EC2 instance running nginx). The proxy component will have a fixed IP address, therefore enabling the use of an IP allowlist to secure your backend service. The communication between I/O Runtime and the proxy component will be secured via mutual TLS (mTLS) communication. 
 
-* Using Mutual TLS (mTLS) between I/O Runtime and an external service. 
+The following steps outline how to:
+* Configure the NGINX proxy component to support mutual TLS (mTLS)
+* Configure an AppBuilder action to use mTLS to securely communicate with the proxy component
+
+_**Prerequisite:** An EC2 instance with NGINX installed. The [official NGINX documentation](https://docs.nginx.com/nginx/deployment-guides/amazon-web-services/ec2-instances-for-nginx/) has more information._
+
+1. Verify SSH connectivity to the EC2 instance. (screenshot of terminal/template cmd)
+    ```
+    ssh -i <your-key.pem> ec2-user@<EC2-IPAddress>
+    ```
+2. Generate certificates needed for mTLS (link out to example-mtls project for generating secrets)
+    * Generate **mtls_server.key/.crt**
+        ```
+        openssl req -x509 -nodes -days 3650 -newkey rsa:4096 -keyout mtls_server.key -out mtls_server.crt
+        ```
+    * Generate **mtls_client.key/.crt**
+        ```
+        openssl req -x509 -nodes -days 3650 -newkey rsa:4096 -keyout mtls_client.key -out mtls_client.crt
+        ```
+3. Use the referenced file `mtls.conf.example` and replace `DESTINATION_HOST` with the final destination you would like to proxy to. For example, if your target host is `api.myhost.com` you would search for the following line
+    ```
+    proxy_pass https://DESTINATION_HOST
+    ```
+    and replace it such that it looks like 
+    ```
+    proxy_pass https://api.myhost.com
+    ```
+    Save the resulting file locally as `mtls.conf` (in the same folder as your certificates).
+
+4. Copy sample NGINX configuration to EC2 instance with updated placeholder details. (terminal screenshots for `scp` cmds)
+    1. First copy files to home folder
+        ```
+        scp -i <your-key.pem> mtls_server.key mtls_server.crt mtls_client.key mtls_client.crt mtls.conf ec2-user@<EC2-IPAddress>:~/
+        ```
+    2. Then move them into place (while connected via SSH to the ec2 instance)
+        ```
+        sudo mv ~/mtls* /etc/nginx/conf.d/
+        ```
+5. Restart `nginx`
+6. Verify you can connect via curl locally from the ec2 instance
+    ```
+    $ curl -ki --cert /etc/nginx/conf.d/mtls_client.crt --key /etc/nginx/conf.d/mtls_client.key https://localhost/
+    ```
+7. In your AppBuilder app, you will need to make changes to wire the mTLS client key and certificate
```

Comment:
Should we add a 6.1 step, which encourages the customer to create an AMI with the configuration, so that they can always use it and changes don't get lost?